### ISSUE TITLE
Server version is different than client version ( connect ECONNREFUSED )

### ISSUE THREAD

**Issue by halloalexkern** [facebook/nuclide#368]
### Issue and Steps to Reproduce
Nuclide was running fine.
Then, I updated my Atom Nuclide package. After that i updated my node-package on the server.
I killed all nuclide processes.
Now i'm on 0.117.0 on both an i get the error message:

Server version is different than client version
Original error message:
connect ECONNREFUSED xx.xxx.243.140:9090

### Expected Behavior
It should connect normally.

### Actual Behavior
I get an error message (as described before)

### Versions
* Atom: nuclide@0.117.0
* Nuclide:  nuclide@0.117.0
* Client OS: MacOSX (latest)
* Server OS (optional): Ubuntu

Thanks!

:-) AK

**Comment by shushz** [comment]
Could you please verify that the server has started successfully and that the port is not being blocked by a firewall or something similar?
The different version message in this case is misleading - the actual reason here is ECONNREFUSED xx.xxx.243.140:9090.

**Comment by tominkozhimala** [comment]
@shushz I'm seeing this same issue. Was running 0.117.0 server and client earlier today with access to all of my files. After I shut down nuclide server I haven't been able to connect again (error: `connect ECONNREFUSED`)

**Comment by shushz** [comment]
@maksimsco what happens if you run: 
curl -X POST http://<YourServerIP>:9090/heartbeat

**Comment by maksimsco** [comment]
Do you mean curl it from client? 

1. Connect from atom
2. curl returns: curl: (7) Failed to connect to 192.168.0.103 port 9090: Connection refused
3. Manually start nuclide-start-server on server
4. curls returns: curl: (52) Empty reply from server
5. Connect from atom
6. curl returns: curl: (7) Failed to connect to 192.168.0.103 port 9090: Connection refused

**Comment by jjh42** [comment]
I am having the same symptoms. Manually starting the server starts a running process, I can telnet to 9090 and curl -X POST http://server:9090/heartbeat gives `curl: (52) Empty reply from server`.

However, when I try and launch through Atom, on the server I briefly see the node process launch, it creates logs in /tmp/ etc. but then quickly disappears. The nuclide logs on the server don't report any error.

Atom 1.5.3 and Nuclide 0.117.0.

**Comment by jjh42** [comment]
Two additional data points.

Downgrading client and server to 0.116.0, and everything seems to work fine again. I was wondering if it might be to do with the change from using an interactive shell to exec (https://github.com/facebook/nuclide/blob/cbc5c62dfe66e73932d195edc886cb6718adefe1/pkg/nuclide/remote-connection/lib/SshHandshake.js)

In doing so, I also found a workaround (although I don't quite understand how this works).

On my laptop I run
`TERM=nuclide ssh $SERVER nuclide-start-server --common-name=$SERVER --workspace=$HOME --json-output-file=/tmp/nuclide-sshhandshake-me -t 60`

and then when I connect through Atom to the same server it seems to attach to the existing server process (and the two server log files show the same certificate being used).

**Comment by shushz** [comment]
@jjh42 Thanks investigating it!
@hansonw what do you think? Could https://github.com/facebook/nuclide/commit/cbc5c62dfe66e73932d195edc886cb6718adefe1 cause this behavior?

**Comment by hansonw** [comment]
hmm... definitely possible. Super weird that this somehow breaks "nohup" - I'll dig into it tomorrow, and revert it if I can't figure out a fix.

**Comment by shushz** [comment]
Fixed by https://github.com/facebook/nuclide/commit/3d49ff69c57bdd370fdfaa31001383d77c28d28a

**Comment by jjh42** [comment]
Thanks!

**Comment by bcarroll22** [comment]
I've still got issues here, I get an ECONNRESET whenever I try to connect

**Comment by shushz** [comment]
The fix is in the repository, it will be a part of the next release - v0.119.0

**Comment by corbt** [comment]
I think I'm still running into this issue even with v0.119.0.

Server version: v0.119.0
Client version: v0.119.0

When I try to connect, I get the following error in my editor's debug pane:
```
[2016-02-25T12:31:20.675Z] [ERROR] nuclide - SshHandshake failed: SERVER_VERSION_MISMATCH, Connection check failed Error: read ECONNRESET
  at exports._errnoException (util.js:837:11)
  at TLSWrap.onread (net.js:544:26)
```

And the following logs on the server in `/tmp/nuclide-vagrant-logs/`
```
2016-02-25 12:30:07,048 - NuclideServer - INFO - Verified Nuclide started on port 9090.
2016-02-25 12:31:14,302 - root - WARNING - Failed to clean up old core dumps ()
2016-02-25 12:31:14,303 - NuclideServerManager - INFO - Trying to determine the port to use for Nuclide server...
2016-02-25 12:31:14,304 - NuclideServerManager - ERROR - The user specified port 9090.
2016-02-25 12:31:14,306 - NuclideServerManager - INFO - Initialized NuclideServer.
2016-02-25 12:31:14,314 - NuclideServerManager - INFO - Starting NuclideServer...
2016-02-25 12:31:14,314 - NuclideServerManager - INFO - Using https.
2016-02-25 12:31:14,315 - NuclideServerManager - INFO - Checking if certificate dir /home/vagrant/.certs exists.
2016-02-25 12:31:14,503 - NuclideServerManager - INFO - Initialized NuclideCertificatesGenerator with common_name: 127.0.0.1
2016-02-25 12:31:14,504 - NuclideServer - INFO - NuclideServer start/restarting with the following arguments:
                           timeout: 60
                           cert:    /home/vagrant/.certs/nuclide.Wjo1vn.server.crt
                           key:     /home/vagrant/.certs/nuclide.Wjo1vn.server.key
                           ca:      /home/vagrant/.certs/nuclide.Wjo1vn.ca.crt
                           force:   False
                           quiet:   False
                           debug:   False
2016-02-25 12:31:14,512 - NuclideServer - INFO - Opening node server subprocess.
2016-02-25 12:31:14,513 - NuclideServer - INFO - Trying to ping the Nuclide server...
2016-02-25 12:31:20,601 - NuclideServer - INFO - Verified Nuclide started on port 9090.
```

Port 9090 is exposed, and I can `curl -X POST http://127.0.0.1:9090/heartbeat` from both the client and the server and get the response `curl: (52) Empty reply from server`.

**Comment by gwing33** [comment]
I am seeing the same issues using v0.119.0 on client/server. I was running it just fine on v0.118.0, but this morning something happened and now I can't connect, I've uninstalled, reinstalled both v0.119.0 and v0.118.0 and to no avail.

**Comment by shushz** [comment]
It seems like a different issue.  I can't reproduce the ECONNRESET. Is there anything else in the logs on either side?

**Comment by bcarroll22** [comment]
Yeah I've got the same issue. Still telling me client/server are different versions. Where are the logs on both sides?

**Comment by gwing33** [comment]
@shushz Here are the client log on my machine (filtered out ip/servernames).
I'm sure it was from various times of trying, so I'm not sure what is/is not important.

```
2016-02-25 09:40:56.466 WARN (pid:679) nuclide - Failed to reuse connectionConfiguration for [servername] { [Error: connect ECONNREFUSED x.x.x.x:8000]
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: 'x.x.x.x',
  port: 8000,
  stackTrace: 
   [ { functionName: 'exports._errnoException',
       methodName: '_errnoException',
       fileName: 'util.js',
       lineNumber: 837,
       columnNumber: 11,
       evalOrigin: 'util.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'exports._exceptionWithHostPort',
       methodName: null,
       fileName: 'util.js',
       lineNumber: 860,
       columnNumber: 20,
       evalOrigin: 'util.js',
       isTopLevel: true,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'afterConnect',
       methodName: 'oncomplete',
       fileName: 'net.js',
       lineNumber: 1060,
       columnNumber: 14,
       evalOrigin: 'net.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false } ] }
Error: connect ECONNREFUSED x.x.x.x:8000
  at Object.exports._errnoException (util.js:837:11)
  at exports._exceptionWithHostPort (util.js:860:20)
  at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1060:14)

2016-02-25 09:41:05.604 DEBUG (pid:679) nuclide - Registering 3.0 service ArcanistBaseService...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias ArcDiagnostic...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering 3.0 service BuckProject...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.605 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias dontRunOptions...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias doRunOptions...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias BuckRunOptions...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering interface BuckProject.
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering 3.0 service FileSystemService...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering type alias FileWithStats...
2016-02-25 09:41:05.606 DEBUG (pid:679) nuclide - Registering 3.0 service ClangService...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.607 DEBUG (pid:679) nuclide - Registering type alias ClangCursorType...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangCursorExtent...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangCompileResult...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangCompletion...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangCompletionsResult...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangDeclarationResult...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangDeclaration...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias ClangDeclarationInfoResult...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering 3.0 service FlowService...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.608 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias Diagnostics...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias Diagnostic...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias SingleMessage...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias Loc...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias ServerStatusType...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias FlowStartLocation...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias FlowOutlineTree...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering 3.0 service FuzzyFileSearchService...
2016-02-25 09:41:05.609 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias FileSearchResult...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering 3.0 service FuzzyFileSearchRxService...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.610 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias FileSearchResultRx...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering 3.0 service BuckUtils...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.611 DEBUG (pid:679) nuclide - Registering interface BuckUtils.
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering 3.0 service MerlinService...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering 3.0 service InfoService...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.612 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering 3.0 service FindInProjectService...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias search$Match...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering type alias search$FileResult...
2016-02-25 09:41:05.613 DEBUG (pid:679) nuclide - Registering 3.0 service HhvmDebuggerProxyService...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias ConnectionConfig...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias NotificationMessage...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering interface HhvmDebuggerProxyService.
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering 3.0 service FileWatcherService...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.614 DEBUG (pid:679) nuclide - Registering type alias watcher$WatchResult...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering 3.0 service SourceControlService...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias HgRepositoryDescription...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering 3.0 service HackService...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.615 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias SymbolTypeValue...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackDiagnosticsResult...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackDiagnostic...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias SingleHackMessage...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackFunctionDetails...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackCompletion...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackCompletionsResult...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackDefinitionResult...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackReferencesResult...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackSearchPosition...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering type alias HackReference...
2016-02-25 09:41:05.616 DEBUG (pid:679) nuclide - Registering 3.0 service HgService...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias StatusCodeIdValue...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias LineDiff...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias DiffInfo...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias RevisionFileCopy...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias RevisionFileChanges...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias RevisionInfo...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering type alias AsyncExecuteRet...
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering interface HgService.
2016-02-25 09:41:05.617 DEBUG (pid:679) nuclide - Registering 3.0 service CtagsService...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias CtagsResult...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering interface CtagsService.
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering 3.0 service LLDBDebuggerRpcService...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias Object...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias Date...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias RegExp...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias Buffer...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias fs.Stats...
2016-02-25 09:41:05.618 DEBUG (pid:679) nuclide - Registering type alias NuclideUri...
2016-02-25 09:41:05.691 ERROR (pid:679) nuclide - SshHandshake failed: SERVER_VERSION_MISMATCH, Connection check failed { [Error: connect ECONNREFUSED x.x.x.x:8000]
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: 'x.x.x.x',
  port: 8000,
  stackTrace: 
   [ { functionName: 'exports._errnoException',
       methodName: '_errnoException',
       fileName: 'util.js',
       lineNumber: 837,
       columnNumber: 11,
       evalOrigin: 'util.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'exports._exceptionWithHostPort',
       methodName: null,
       fileName: 'util.js',
       lineNumber: 860,
       columnNumber: 20,
       evalOrigin: 'util.js',
       isTopLevel: true,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'afterConnect',
       methodName: 'oncomplete',
       fileName: 'net.js',
       lineNumber: 1060,
       columnNumber: 14,
       evalOrigin: 'net.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false } ] }
Error: connect ECONNREFUSED x.x.x.x:8000
  at Object.exports._errnoException (util.js:837:11)
  at exports._exceptionWithHostPort (util.js:860:20)
  at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1060:14)

2016-02-25 09:41:05.701 DEBUG (pid:679) nuclide - { [Error: connect ECONNREFUSED x.x.x.x:8000]
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: 'x.x.x.x',
  port: 8000,
  stackTrace: 
   [ { functionName: 'exports._errnoException',
       methodName: '_errnoException',
       fileName: 'util.js',
       lineNumber: 837,
       columnNumber: 11,
       evalOrigin: 'util.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'exports._exceptionWithHostPort',
       methodName: null,
       fileName: 'util.js',
       lineNumber: 860,
       columnNumber: 20,
       evalOrigin: 'util.js',
       isTopLevel: true,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'afterConnect',
       methodName: 'oncomplete',
       fileName: 'net.js',
       lineNumber: 1060,
       columnNumber: 14,
       evalOrigin: 'net.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false } ] }
Error: connect ECONNREFUSED x.x.x.x:8000
  at Object.exports._errnoException (util.js:837:11)
  at exports._exceptionWithHostPort (util.js:860:20)
  at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1060:14)
```

**Comment by shushz** [comment]
@gwing33 Notice the different error code in the logs you provided. It's ECONNREFUSED, which means the server doesn't run at the time the connection attempt was made. It's different from the issue @corbt is having, where the server did run and Nuclide connected to it successfully, but then one of the sides (probably the server) has closed the connection.

@bcarroll22 The different versions message is misleading. It's shown when there's a connectivity problem between Nuclide and the server. The detail of the message show what kind of a problem really occurred.
BTW, the logs are at /tmp/nuclide-<username>-logs directory on both the server and the client.

**Comment by corbt** [comment]
@shushz Ok, here are more-full logs:

### Client
```
2016-02-25 19:02:46.007 DEBUG (pid:433) nuclide - Registering 3.0 service ArcanistBaseService...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering type alias ArcDiagnostic...
2016-02-25 19:02:46.111 DEBUG (pid:433) nuclide - Registering 3.0 service BuckProject...
2016-02-25 19:02:46.274 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.274 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.274 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.274 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.274 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.274 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.275 DEBUG (pid:433) nuclide - Registering type alias dontRunOptions...
2016-02-25 19:02:46.275 DEBUG (pid:433) nuclide - Registering type alias doRunOptions...
2016-02-25 19:02:46.275 DEBUG (pid:433) nuclide - Registering type alias BuckRunOptions...
2016-02-25 19:02:46.275 DEBUG (pid:433) nuclide - Registering interface BuckProject.
2016-02-25 19:02:46.275 DEBUG (pid:433) nuclide - Registering 3.0 service FileSystemService...
2016-02-25 19:02:46.388 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.388 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.388 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.388 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.388 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.389 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.389 DEBUG (pid:433) nuclide - Registering type alias FileWithStats...
2016-02-25 19:02:46.389 DEBUG (pid:433) nuclide - Registering 3.0 service ClangService...
2016-02-25 19:02:46.450 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.450 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangCursorType...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangCursorExtent...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangCompileResult...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangCompletion...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangCompletionsResult...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangDeclarationResult...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangDeclaration...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering type alias ClangDeclarationInfoResult...
2016-02-25 19:02:46.451 DEBUG (pid:433) nuclide - Registering 3.0 service FlowService...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias Diagnostics...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias Diagnostic...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias SingleMessage...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias Loc...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias ServerStatusType...
2016-02-25 19:02:46.489 DEBUG (pid:433) nuclide - Registering type alias FlowStartLocation...
2016-02-25 19:02:46.490 DEBUG (pid:433) nuclide - Registering type alias FlowOutlineTree...
2016-02-25 19:02:46.490 DEBUG (pid:433) nuclide - Registering 3.0 service FuzzyFileSearchService...
2016-02-25 19:02:46.502 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.502 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.502 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.502 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.502 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.502 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.503 DEBUG (pid:433) nuclide - Registering type alias FileSearchResult...
2016-02-25 19:02:46.503 DEBUG (pid:433) nuclide - Registering 3.0 service FuzzyFileSearchRxService...
2016-02-25 19:02:46.514 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering type alias FileSearchResultRx...
2016-02-25 19:02:46.515 DEBUG (pid:433) nuclide - Registering 3.0 service BuckUtils...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering interface BuckUtils.
2016-02-25 19:02:46.525 DEBUG (pid:433) nuclide - Registering 3.0 service MerlinService...
2016-02-25 19:02:46.550 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.550 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.550 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.550 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.550 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.551 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.551 DEBUG (pid:433) nuclide - Registering 3.0 service InfoService...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.559 DEBUG (pid:433) nuclide - Registering 3.0 service FindInProjectService...
2016-02-25 19:02:46.569 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias search$Match...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering type alias search$FileResult...
2016-02-25 19:02:46.570 DEBUG (pid:433) nuclide - Registering 3.0 service HhvmDebuggerProxyService...
2016-02-25 19:02:46.599 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias ConnectionConfig...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering type alias NotificationMessage...
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering interface HhvmDebuggerProxyService.
2016-02-25 19:02:46.600 DEBUG (pid:433) nuclide - Registering 3.0 service FileWatcherService...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering type alias watcher$WatchResult...
2016-02-25 19:02:46.627 DEBUG (pid:433) nuclide - Registering 3.0 service SourceControlService...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.633 DEBUG (pid:433) nuclide - Registering type alias HgRepositoryDescription...
2016-02-25 19:02:46.634 DEBUG (pid:433) nuclide - Registering 3.0 service HackService...
2016-02-25 19:02:46.694 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.695 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias SymbolTypeValue...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias HackDiagnosticsResult...
2016-02-25 19:02:46.697 DEBUG (pid:433) nuclide - Registering type alias HackDiagnostic...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias SingleHackMessage...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackFunctionDetails...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackCompletion...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackCompletionsResult...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackDefinitionResult...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackReferencesResult...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackSearchPosition...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering type alias HackReference...
2016-02-25 19:02:46.698 DEBUG (pid:433) nuclide - Registering 3.0 service HgService...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias StatusCodeIdValue...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias LineDiff...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias DiffInfo...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias RevisionFileCopy...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias RevisionFileChanges...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias RevisionInfo...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering type alias AsyncExecuteRet...
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering interface HgService.
2016-02-25 19:02:46.743 DEBUG (pid:433) nuclide - Registering 3.0 service CtagsService...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering type alias CtagsResult...
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering interface CtagsService.
2016-02-25 19:02:46.762 DEBUG (pid:433) nuclide - Registering 3.0 service LLDBDebuggerRpcService...
2016-02-25 19:02:46.773 DEBUG (pid:433) nuclide - Registering type alias Object...
2016-02-25 19:02:46.773 DEBUG (pid:433) nuclide - Registering type alias Date...
2016-02-25 19:02:46.773 DEBUG (pid:433) nuclide - Registering type alias RegExp...
2016-02-25 19:02:46.773 DEBUG (pid:433) nuclide - Registering type alias Buffer...
2016-02-25 19:02:46.774 DEBUG (pid:433) nuclide - Registering type alias fs.Stats...
2016-02-25 19:02:46.774 DEBUG (pid:433) nuclide - Registering type alias NuclideUri...
2016-02-25 19:02:46.787 ERROR (pid:433) nuclide - SshHandshake failed: SERVER_VERSION_MISMATCH, Connection check failed { [Error: read ECONNRESET]
  code: 'ECONNRESET',
  errno: 'ECONNRESET',
  syscall: 'read',
  stackTrace:
   [ { functionName: 'exports._errnoException',
       methodName: null,
       fileName: 'util.js',
       lineNumber: 837,
       columnNumber: 11,
       evalOrigin: 'util.js',
       isTopLevel: true,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'onread',
       methodName: null,
       fileName: 'net.js',
       lineNumber: 544,
       columnNumber: 26,
       evalOrigin: 'net.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false } ] }
Error: read ECONNRESET
  at exports._errnoException (util.js:837:11)
  at TLSWrap.onread (net.js:544:26)

2016-02-25 19:02:46.811 DEBUG (pid:433) nuclide - { [Error: read ECONNRESET]
  code: 'ECONNRESET',
  errno: 'ECONNRESET',
  syscall: 'read',
  stackTrace:
   [ { functionName: 'exports._errnoException',
       methodName: null,
       fileName: 'util.js',
       lineNumber: 837,
       columnNumber: 11,
       evalOrigin: 'util.js',
       isTopLevel: true,
       isEval: false,
       isNative: false,
       isConstructor: false },
     { functionName: 'onread',
       methodName: null,
       fileName: 'net.js',
       lineNumber: 544,
       columnNumber: 26,
       evalOrigin: 'net.js',
       isTopLevel: false,
       isEval: false,
       isNative: false,
       isConstructor: false } ] }
Error: read ECONNRESET
  at exports._errnoException (util.js:837:11)
  at TLSWrap.onread (net.js:544:26)
```

### Server
```
==> /tmp/nuclide-vagrant-logs/server-start/nuclide.log <==
2016-02-25 19:02:30,632 - root - WARNING - Failed to clean up old core dumps ()
2016-02-25 19:02:30,633 - NuclideServerManager - INFO - Trying to determine the port to use for Nuclide server...
2016-02-25 19:02:30,634 - NuclideServerManager - ERROR - The user specified port 9090.
2016-02-25 19:02:30,635 - NuclideServerManager - INFO - Initialized NuclideServer.
2016-02-25 19:02:30,714 - NuclideServerManager - INFO - A Nuclide server is already running.                           Running version: 119. Desired version: 119.
2016-02-25 19:02:30,723 - NuclideServerManager - INFO - Restarting Nuclide server on port 9090
2016-02-25 19:02:30,729 - NuclideServer - INFO - Stopped old Nuclide server on port 9090.
2016-02-25 19:02:30,730 - NuclideServerManager - INFO - Starting NuclideServer...
2016-02-25 19:02:30,732 - NuclideServerManager - INFO - Using https.
2016-02-25 19:02:30,733 - NuclideServerManager - INFO - Checking if certificate dir /home/vagrant/.certs exists.
2016-02-25 19:02:30,857 - NuclideServerManager - INFO - Initialized NuclideCertificatesGenerator with common_name: 127.0.0.1
2016-02-25 19:02:30,857 - NuclideServer - INFO - NuclideServer start/restarting with the following arguments:
                           timeout: 60
                           cert:    /home/vagrant/.certs/nuclide.c3h1dA.server.crt
                           key:     /home/vagrant/.certs/nuclide.c3h1dA.server.key
                           ca:      /home/vagrant/.certs/nuclide.c3h1dA.ca.crt
                           force:   False
                           quiet:   False
                           debug:   False
2016-02-25 19:02:30,865 - NuclideServer - INFO - Opening node server subprocess.
2016-02-25 19:02:30,867 - NuclideServer - INFO - Trying to ping the Nuclide server...
2016-02-25 19:02:34,936 - NuclideServer - INFO - Verified Nuclide started on port 9090.
```

I can open a new issue for this if that would be helpful.

**Comment by shushz** [comment]
@corbt Yes, please do - it seems like a different issue.
Also, could please you attach all of the server logs. Those you attached are only from the server startup process. For some reason there seems to be a level of inconsistency about log placing. Please run this command on the server to find all of the log files:
```bash
ps aux | grep nuclide-main | grep -v grep | tr -s ' ' | cut -d ' ' -f2 | xargs lsof -p | grep logs
```

**Comment by gwing33** [comment]
@shushz I will keep on this thread since I'm seemingly getting similar errors as mentioned in this thread.

When I run `nuclide-start-server -p 8000` on the server, everything starts up fine.

If I curl from my client I get `curl: (52) Empty reply from server`.

As soon as I try to connect through atom, the server dies (I.E. `ps aux | grep nuclide` yields no results, or if I curl I get `curl: (7) Failed to connect to x.x.x.x port 8000: Connection refused`).

Any timestamp around 20:59:14 is from my manual run, 21:01:09 on is from atom.

```
$ cat /tmp/nuclide-gerald.leenerts-logs/server-start/nuclide-2016-02-25T20:59:11.nohup.out
...
[2016-02-25 20:59:14.738] [DEBUG] nuclide - Registering function CtagsService/getCtagsService...
[2016-02-25 20:59:14.738] [DEBUG] nuclide - Registering 3.0 service LLDBDebuggerRpcService...
[2016-02-25 20:59:14.781] [DEBUG] nuclide - Registering type alias Object...
[2016-02-25 20:59:14.781] [DEBUG] nuclide - Registering type alias Date...
[2016-02-25 20:59:14.781] [DEBUG] nuclide - Registering type alias RegExp...
[2016-02-25 20:59:14.781] [DEBUG] nuclide - Registering type alias Buffer...
[2016-02-25 20:59:14.781] [DEBUG] nuclide - Registering type alias fs.Stats...
[2016-02-25 20:59:14.781] [DEBUG] nuclide - Registering type alias NuclideUri...
[2016-02-25 20:59:14.799] [INFO] nuclide - NuclideServer started on port 8000.
[2016-02-25 20:59:14.808] [INFO] nuclide - NuclideServer event loop blocked for 1235ms
[2016-02-25T21:01:09.211Z] [INFO] nuclide - NuclideServer event loop blocked for 99ms
```

Also you can see in the logs the first start I manually ran, and then the 2nd start by atom.
The only thing that stands out here is the `NuclideCertificatesGenerator` names are different.

```
$ cat /tmp/nuclide-gerald.leenerts-logs/server-start/nuclide.log
2016-02-25 20:59:11,403 - root - WARNING - Failed to clean up old core dumps ()
2016-02-25 20:59:11,403 - NuclideServerManager - INFO - Trying to determine the port to use for Nuclide server...
2016-02-25 20:59:11,403 - NuclideServerManager - ERROR - The user specified port 8000.
2016-02-25 20:59:11,403 - NuclideServerManager - INFO - Initialized NuclideServer.
2016-02-25 20:59:11,418 - NuclideServerManager - INFO - Starting NuclideServer...
2016-02-25 20:59:11,418 - NuclideServerManager - INFO - Using https.
2016-02-25 20:59:11,419 - NuclideServerManager - INFO - Checking if certificate dir /home/gerald.leenerts/.certs exists.
2016-02-25 20:59:11,653 - NuclideServerManager - INFO - Initialized NuclideCertificatesGenerator with common_name: gerald.leenerts.nuclide
2016-02-25 20:59:11,653 - NuclideServer - INFO - NuclideServer start/restarting with the following arguments:
                           timeout: 10
                           cert:    /home/gerald.leenerts/.certs/nuclide.o5RJxa.server.crt
                           key:     /home/gerald.leenerts/.certs/nuclide.o5RJxa.server.key
                           ca:      /home/gerald.leenerts/.certs/nuclide.o5RJxa.ca.crt
                           force:   False
                           quiet:   False
                           debug:   False
2016-02-25 20:59:11,668 - NuclideServer - INFO - Opening node server subprocess.
2016-02-25 20:59:11,669 - NuclideServer - INFO - Trying to ping the Nuclide server...
2016-02-25 20:59:15,803 - NuclideServer - INFO - Verified Nuclide started on port 8000.
2016-02-25 21:01:14,546 - root - WARNING - Failed to clean up old core dumps ()
2016-02-25 21:01:14,547 - NuclideServerManager - INFO - Trying to determine the port to use for Nuclide server...
2016-02-25 21:01:14,547 - NuclideServerManager - ERROR - The user specified port 8000.
2016-02-25 21:01:14,547 - NuclideServerManager - INFO - Initialized NuclideServer.
2016-02-25 21:01:14,611 - NuclideServerManager - INFO - A Nuclide server is already running.                           Running version: 119. Desired version: 119.
2016-02-25 21:01:14,634 - NuclideServerManager - INFO - Restarting Nuclide server on port 8000
2016-02-25 21:01:14,639 - NuclideServer - INFO - Stopped old Nuclide server on port 8000.
2016-02-25 21:01:14,639 - NuclideServerManager - INFO - Starting NuclideServer...
2016-02-25 21:01:14,639 - NuclideServerManager - INFO - Using https.
2016-02-25 21:01:14,639 - NuclideServerManager - INFO - Checking if certificate dir /home/gerald.leenerts/.certs exists.
2016-02-25 21:01:15,005 - NuclideServerManager - INFO - Initialized NuclideCertificatesGenerator with common_name: completely.diff.name
2016-02-25 21:01:15,005 - NuclideServer - INFO - NuclideServer start/restarting with the following arguments:
                           timeout: 60
                           cert:    /home/gerald.leenerts/.certs/nuclide.4265X_.server.crt
                           key:     /home/gerald.leenerts/.certs/nuclide.4265X_.server.key
                           ca:      /home/gerald.leenerts/.certs/nuclide.4265X_.ca.crt
                           force:   False
                           quiet:   False
                           debug:   False
2016-02-25 21:01:15,021 - NuclideServer - INFO - Opening node server subprocess.
2016-02-25 21:01:15,022 - NuclideServer - INFO - Trying to ping the Nuclide server...
2016-02-25 21:01:18,235 - NuclideServer - INFO - Verified Nuclide started on port 8000.
```

All of this being said, if I roll back to v0.116.0 things are working fine.

If there is any information I can give you, please let me know, or if this is a different thing from this issue, I'm happy to open a new ticket.

**Comment by shushz** [comment]
OK. Seems I'm able to reproduce it on the 119, but not on the master

**Comment by shushz** [comment]
I just learned that the fix I thought was in 119 in fact wasn't there. All of the issues in this thread were, in all likelihood, introduced by a change in v0.117.0 and were already fixed by https://github.com/facebook/nuclide/commit/3d49ff69c57bdd370fdfaa31001383d77c28d28a
which will be part of the v0.120.0 release.

Sorry for the misunderstanding.

**Comment by gwing33** [comment]
Thanks @shushz good to know!

**Comment by nmote** [comment]
I just pushed v0.120.0 which includes this fix.

**Comment by bcarroll22** [comment]
Still have the same error, actually. Updated both client and server to 0.120

**Comment by shushz** [comment]
@bcarroll22 Damn! That might be a different issue then. Only you and @corbt had the ECONNRESET error.
@gwing33, @corbt, @halloalexkern, @maksimsco what about you? Did the v0.120.0 fix that for you?

**Comment by gwing33** [comment]
@shushz v0.120.0 fixed my issue. Thanks!

**Comment by bcarroll22** [comment]
Yeah mine is actually fixed too, it was a port issue that I had overlooked.
Thanks!
